{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "cellView": "form",
        "id": "II6F7ThkI10I",
        "outputId": "d913a240-cf7d-4876-c595-319cbb149473",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "                                                                                                    \n",
            "ERROR:  authentication failed: The authtoken you specified is properly formed, but it is invalid.\n",
            "ERROR:  Your authtoken: 2rLuTWX0s5U4xJTTvIR84FWu8em_5kZVT1SfxFP8vjuzj5Y7o\n",
            "ERROR:  This usually happens when:\n",
            "ERROR:      - You reset your authtoken\n",
            "ERROR:      - Your authtoken was for a team account that you were removed from\n",
            "ERROR:      - You are using ngrok link and this credential was explicitly revoked\n",
            "ERROR:  Go to your ngrok dashboard and double check that your authtoken is correct:\n",
            "ERROR:  https://dashboard.ngrok.com/get-started/your-authtoken\n",
            "ERROR:  \n",
            "ERROR:  ERR_NGROK_107\n",
            "ERROR:  https://ngrok.com/docs/errors/err_ngrok_107\n",
            "ERROR:  \n",
            "t=2025-01-08T15:46:29+0000 lvl=eror msg=\"failed to reconnect session\" obj=tunnels.session err=\"authentication failed: The authtoken you specified is properly formed, but it is invalid.\\nYour authtoken: 2rLuTWX0s5U4xJTTvIR84FWu8em_5kZVT1SfxFP8vjuzj5Y7o\\nThis usually happens when:\\n    - You reset your authtoken\\n    - Your authtoken was for a team account that you were removed from\\n    - You are using ngrok link and this credential was explicitly revoked\\nGo to your ngrok dashboard and double check that your authtoken is correct:\\nhttps://dashboard.ngrok.com/get-started/your-authtoken\\r\\n\\r\\nERR_NGROK_107\\r\\n\"\n",
            "t=2025-01-08T15:46:29+0000 lvl=eror msg=\"session closing\" obj=tunnels.session err=\"authentication failed: The authtoken you specified is properly formed, but it is invalid.\\nYour authtoken: 2rLuTWX0s5U4xJTTvIR84FWu8em_5kZVT1SfxFP8vjuzj5Y7o\\nThis usually happens when:\\n    - You reset your authtoken\\n    - Your authtoken was for a team account that you were removed from\\n    - You are using ngrok link and this credential was explicitly revoked\\nGo to your ngrok dashboard and double check that your authtoken is correct:\\nhttps://dashboard.ngrok.com/get-started/your-authtoken\\r\\n\\r\\nERR_NGROK_107\\r\\n\"\n",
            "t=2025-01-08T15:46:29+0000 lvl=eror msg=\"terminating with error\" obj=app err=\"authentication failed: The authtoken you specified is properly formed, but it is invalid.\\nYour authtoken: 2rLuTWX0s5U4xJTTvIR84FWu8em_5kZVT1SfxFP8vjuzj5Y7o\\nThis usually happens when:\\n    - You reset your authtoken\\n    - Your authtoken was for a team account that you were removed from\\n    - You are using ngrok link and this credential was explicitly revoked\\nGo to your ngrok dashboard and double check that your authtoken is correct:\\nhttps://dashboard.ngrok.com/get-started/your-authtoken\\r\\n\\r\\nERR_NGROK_107\\r\\n\"\n",
            "t=2025-01-08T15:46:29+0000 lvl=crit msg=\"command failed\" err=\"authentication failed: The authtoken you specified is properly formed, but it is invalid.\\nYour authtoken: 2rLuTWX0s5U4xJTTvIR84FWu8em_5kZVT1SfxFP8vjuzj5Y7o\\nThis usually happens when:\\n    - You reset your authtoken\\n    - Your authtoken was for a team account that you were removed from\\n    - You are using ngrok link and this credential was explicitly revoked\\nGo to your ngrok dashboard and double check that your authtoken is correct:\\nhttps://dashboard.ngrok.com/get-started/your-authtoken\\r\\n\\r\\nERR_NGROK_107\\r\\n\"\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/bin/autotrain\", line 8, in <module>\n",
            "    sys.exit(main())\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/autotrain/cli/autotrain.py\", line 68, in main\n",
            "    command.run()\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/autotrain/cli/run_app.py\", line 127, in run\n",
            "    active_tunnels = ngrok.get_tunnels()\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/pyngrok/ngrok.py\", line 394, in get_tunnels\n",
            "    api_url = get_ngrok_process(pyngrok_config).api_url\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/pyngrok/ngrok.py\", line 156, in get_ngrok_process\n",
            "    return process.get_process(pyngrok_config)\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/pyngrok/process.py\", line 235, in get_process\n",
            "    return _start_process(pyngrok_config)\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/pyngrok/process.py\", line 398, in _start_process\n",
            "    raise PyngrokNgrokError(f\"The ngrok process errored on start: {ngrok_process.startup_error}.\",\n",
            "pyngrok.exception.PyngrokNgrokError: The ngrok process errored on start: authentication failed: The authtoken you specified is properly formed, but it is invalid.\\nYour authtoken: 2rLuTWX0s5U4xJTTvIR84FWu8em_5kZVT1SfxFP8vjuzj5Y7o\\nThis usually happens when:\\n    - You reset your authtoken\\n    - Your authtoken was for a team account that you were removed from\\n    - You are using ngrok link and this credential was explicitly revoked\\nGo to your ngrok dashboard and double check that your authtoken is correct:\\nhttps://dashboard.ngrok.com/get-started/your-authtoken\\r\\n\\r\\nERR_NGROK_107\\r\\n.\n"
          ]
        }
      ],
      "source": [
        "#@title ðŸ¤— AutoTrain\n",
        "#@markdown In order to use this colab\n",
        "#@markdown - Enter your [Hugging Face Write Token](https://huggingface.co/settings/tokens)\n",
        "#@markdown - Enter your [ngrok auth token](https://dashboard.ngrok.com/get-started/your-authtoken)\n",
        "huggingface_token = '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' # @param {type:\"string\"}\n",
        "ngrok_token = \"2rLuTWX0s5U4xJTTvIR84FWu8em_5kZVT1SfxFP8vjuzj5Y7o\" # @param {type:\"string\"}\n",
        "\n",
        "#@markdown\n",
        "#@markdown - Attach appropriate accelerator `Runtime > Change runtime type > Hardware accelerator`\n",
        "#@markdown - click `Runtime > Run all`\n",
        "#@markdown - Follow the link to access the UI\n",
        "#@markdown - Training happens inside this Google Colab\n",
        "#@markdown - report issues / feature requests [here](https://github.com/huggingface/autotrain-advanced/issues)\n",
        "\n",
        "import os\n",
        "os.environ[\"HF_TOKEN\"] = str(huggingface_token)\n",
        "os.environ[\"NGROK_AUTH_TOKEN\"] = str(ngrok_token)\n",
        "os.environ[\"AUTOTRAIN_LOCAL\"] = \"1\"\n",
        "\n",
        "!pip install -U autotrain-advanced > install_logs.txt 2>&1\n",
        "!autotrain app --share"
      ]
    }
  ],
  "metadata": {
    "accelerator": "GPU",
    "colab": {
      "gpuType": "A100",
      "provenance": [],
      "machine_shape": "hm"
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}